File: client-scalaz/src/main/scala/org/openapitools/client/api/OrsonAiTechTuneResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import OrsonAiTechTuneResponse._

case class OrsonAiTechTuneResponse (
  requestId: Option[String],
responseCode: Option[Integer],
responseRaw: Option[String],
people: Option[Integer],
wasCutoff: Option[Boolean],
frames: Option[Integer],
framesWithoutAFace: Option[Integer],
framesWithFaceOffscreen: Option[Integer],
framesWithWrongNumberOfPeople: Option[Integer],
height: Option[Integer],
width: Option[Integer],
fps: Option[Integer])

object OrsonAiTechTuneResponse {
  import DateTimeCodecs._

  implicit val OrsonAiTechTuneResponseCodecJson: CodecJson[OrsonAiTechTuneResponse] = CodecJson.derive[OrsonAiTechTuneResponse]
  implicit val OrsonAiTechTuneResponseDecoder: EntityDecoder[OrsonAiTechTuneResponse] = jsonOf[OrsonAiTechTuneResponse]
  implicit val OrsonAiTechTuneResponseEncoder: EntityEncoder[OrsonAiTechTuneResponse] = jsonEncoderOf[OrsonAiTechTuneResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/orson_ai_tech_tune_response.clj
(ns sirqul-io-t-platform.specs.orson-ai-tech-tune-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def orson-ai-tech-tune-response-data
  {
   (ds/opt :requestId) string?
   (ds/opt :responseCode) int?
   (ds/opt :responseRaw) string?
   (ds/opt :people) int?
   (ds/opt :wasCutoff) boolean?
   (ds/opt :frames) int?
   (ds/opt :framesWithoutAFace) int?
   (ds/opt :framesWithFaceOffscreen) int?
   (ds/opt :framesWithWrongNumberOfPeople) int?
   (ds/opt :height) int?
   (ds/opt :width) int?
   (ds/opt :fps) int?
   })

(def orson-ai-tech-tune-response-spec
  (ds/spec
    {:name ::orson-ai-tech-tune-response
     :spec orson-ai-tech-tune-response-data}))
